circuitpython_nrf24l01/rf24.py

```diff
@@ -217,70 +235,68 @@ def address_length(self, length: int):
-        if self._pipe0_read_addr != address and self._aa & 1:
-            for i, val in enumerate(address):
-                self._pipes[0][i] = val  # type: ignore[assignment, index]
-            self._reg_write_bytes(RX_ADDR_P0, address)
-        for i, val in enumerate(address):
-            self._tx_address[i] = val
-        self._reg_write_bytes(TX_ADDRESS, address)
+        addr_len = min(len(address), self._addr_len)
+        addr = address[:addr_len]
+        self._tx_address[:addr_len] = addr
+        if self._config & 1 == 0 and self._aa & 1:
+            self._reg_write_bytes(_RX_ADDR_P0, addr)
+        self._reg_write_bytes(_TX_ADDR, addr)
 
     def close_rx_pipe(self, pipe_number: int) -> None:
         """Close a specific data pipe from RX transmissions."""
         if pipe_number < 0 or pipe_number > 5:
             raise IndexError("pipe number must be in range [0, 5]")
-        self._open_pipes = self._reg_read(OPEN_PIPES) & ~(1 << pipe_number)
+        self._open_pipes = self._reg_read(_EN_RX) & ~(1 << pipe_number)
         if not pipe_number:
-            self._pipe0_read_addr = None
-        self._reg_write(OPEN_PIPES, self._open_pipes)
+            self._is_p0_rx = False
+        self._reg_write(_EN_RX, self._open_pipes)
 
     def open_rx_pipe(self, pipe_number: int, address: Union[bytes, bytearray]) -> None:
         """Open a specific data pipe for RX transmissions."""
         if not 0 <= pipe_number <= 5:
             raise IndexError("pipe number must be in range [0, 5]")
         if not address:
             raise ValueError("address length cannot be 0")
+        addr_len = min(len(address), self._addr_len)
+        addr = address[:addr_len]
         if pipe_number < 2:
             if not pipe_number:
-                self._pipe0_read_addr = address
-            for i, val in enumerate(address):
-                self._pipes[pipe_number][i] = val  # type: ignore[assignment, index]
-            self._reg_write_bytes(RX_ADDR_P0 + pipe_number, address)
+                self._is_p0_rx = True
+            self._pipes[pipe_number][:addr_len] = addr  # type: ignore[assignment, index]
+            if self._config & 1 or pipe_number != 0:
+                self._reg_write_bytes(_RX_ADDR_P0 + pipe_number, addr)
         else:
-            self._pipes[pipe_number] = address[0]
-            self._reg_write(RX_ADDR_P0 + pipe_number, address[0])
-        self._open_pipes = self._reg_read(OPEN_PIPES) | (1 << pipe_number)
-        self._reg_write(OPEN_PIPES, self._open_pipes)
+            self._pipes[pipe_number] = addr[0]
+            self._reg_write(_RX_ADDR_P0 + pipe_number, address[0])
+        self._open_pipes = self._reg_read(_EN_RX) | (1 << pipe_number)
+        self._reg_write(_EN_RX, self._open_pipes)
 
     @property
     def listen(self) -> bool:
         """This attribute is the primary role as a radio."""
-        return self.power and bool(self._config & 1)
+        return self._config & 3 == 3
```

Comment:
```suggestion
        return bool(self._config & 1)
```
The radio's power state can be determined by the `power` property. Just describe the current radio mode here.